commons/src/main/java/fr/free/nrw/commons/category/PrefixUpdater.java

```diff
@@ -0,0 +1,77 @@
+package fr.free.nrw.commons.category;
+
+import android.os.AsyncTask;
+import android.text.TextUtils;
+import android.util.Log;
+import android.view.View;
+
+import org.mediawiki.api.ApiResult;
+import org.mediawiki.api.MWApi;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import fr.free.nrw.commons.CommonsApplication;
+
+public class PrefixUpdater extends AsyncTask<Void, Void, ArrayList<String>> {
+
+    private String filter;
+    private static final String TAG = PrefixUpdater.class.getName();
+    CategorizationFragment catFragment;
+
+    public PrefixUpdater(CategorizationFragment catFragment) {
```

Comment:
This parameter is fine but a future improvement might be to pass an interface instead of a Fragment or sufficiently separate concerns so only the operates on the final results.